frontend/src/services/fetchData.ts

```diff
@@ -59,7 +61,16 @@ export const deletAndRefreshUserRoleAssignments = async (
 export const fetchApplicationRoles = async (
     applicationId: number | undefined
 ) => {
-    if (!applicationId) return;
```

Comment:
So the original error is cased by this situation, when no application is selected, it just return, and not make any api call, so we can't get the 401 unauthorized error from backend. By logic, when there is no application selected, we should redirect the user back to the dashboard page to let him make a selection. And when he gets redirect to the dashboard page, dashboard page will try to get applications and then return the 401 unauthorized error and redirect back to landing page.